x-pack/plugins/security_solution/common/endpoint/types/index.ts

```diff
@@ -578,7 +642,7 @@ export type ResolverEvent = EndpointEvent | LegacyEndpointEvent;
  * All mappings in Elasticsearch support arrays. They can also return null values or be missing. For example, a `keyword` mapping could return `null` or `[null]` or `[]` or `'hi'`, or `['hi', 'there']`. We need to handle these cases in order to avoid throwing an error.
  * When dealing with an value that comes from ES, wrap the underlying type in `ECSField`. For example, if you have a `keyword` or `text` value coming from ES, cast it to `ECSField<string>`.
  */
-export type ECSField<T> = T | null | Array<T | null>;
+export type ECSField<T> = T | null | undefined | Array<T | null>;
```

Comment:
Needed to add `undefined` here even though ES won't ever return that. This avoids types errors that TS was giving  me.